test/test_thread_utils.cpp

```diff
@@ -6,35 +6,85 @@
 
 namespace mamba
 {
-    TEST(thread_utils, interrupt)
+    namespace
     {
-#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
+        std::mutex res_mutex;
         int res = 0;
```

Comment:
This should be a local variable to the test_interruption_guard function

Reply:
fixde